src/components/timecard/select-timecard-period-type/SelectTimecardPeriodType.jsx

```diff
@@ -48,7 +48,12 @@ const SelectTimecardPeriodType = () => {
       <form
         className="select-timecard-period-type"
         onSubmit={handleSubmit((data) => {
-          setTimecardData({ ...timecardData, timePeriodType: data[radioName] });
+          const newTimeEntries = timeEntries;
+          newTimeEntries.push({
+            timePeriodType: data[radioName],
+            timePeriodTypeId: '00000000-0000-0000-0000-000000000001',
+          });
```

Comment:
Bonus points for making new array in one statement. I did stumble for a minute over the new variable being set to be the same as the old one at first:

```suggestion
          const newTimeEntries [
            ...timeEntries, 
            {
              timePeriodType: data[radioName],
              timePeriodTypeId: '00000000-0000-0000-0000-000000000001',
            }
          ];
```

Reply:
Or snap you could even do this inline in the `setTimeEntries()` function and avoid the variable. In my view it would increase the clarity as you don’t have to wonder if the variable is going to be used for any other purpose, but I’ll acknowledge that this is subjective

Reply:
Yeah that's much cleaner, just updated now